extensions/project-descriptor.md

```diff
@@ -7,81 +7,144 @@ A project descriptor is a file that MAY contain configuration for apps, services
-[metadata]
+[_.metadata]
 # additional arbitrary keys allowed
 ```
 
-The following sections describe each part of the schema in detail.
-
-## `[project]`
-
-The top-level `[project]` table MAY contain configuration about the repository, including `id` and `version`. It MAY also include metadata about how it is authored, documented, and version controlled.
-
-The `project.id`
+The top-level `_` table MAY contain configuration about the repository, including `id` and `version`. It MAY also include metadata about how it is authored, documented, and version controlled. It MUST contain `schema-version`  to denote which schema version the descriptor is using.
 
 ```toml
-[project]
+[_]
+schema-version = "<string>"
 id = "<string>"
 name = "<string>"
 version = "<string>"
+authors = ["<string>"]
+documentation-url = "<uri>"
+source-url = "<uri>"
 ```
 
+* `schema-version` - version identifier for the schema of the `_` table and structure of the project descriptor file.
 * `id` - (optional) the machine readable identifier of the project (ex. "com.example.myservice")
 * `name` - (optional) the human readable name of the project (ex. "My Example Service")
 * `version` - (optional) and arbitrary string representing the version of the project
 * `authors` - (optional) the names and/or email addresses of the project's authors
-* `documentation-url` - (optional) a URL to the documentation for the project
+* `documentation-url` - (optional) a URL to the documentation for the project.
 * `source-url` - (optional) a URL to the source code for the project
 
-## `[[project.licenses]]`
+#### `_.licenses` (optional)
+
+This table MAY contain project licenses.
 
-An optional list of project licenses.
+```toml
+[[_.licenses]]
+type = "<string>"
+uri = "<uri>"
+```
 
 * `type` - This MAY use the [SPDX 2.1 license expression](https://spdx.org/spdx-specification-21-web-version), but is not limited to identifiers in the [SPDX Licenses List](https://spdx.org/licenses/).
 * `uri` - If this project is using a nonstandard license, then this key MAY be specified in lieu of or in addition to `type` to point to the license.
 
-## `[build.include]` and `[build.exclude]`
+#### `_.metadata` (optional)
 
-A optional list of files to include in the build (while excluding everything else):
+This is a free form table for users to use as they see fit. The keys in this table are not validated.
 
 ```toml
-[build]
+[_.metadata.foo]
+checksum = "a28a0d7772df1f918da2b1102da4ff35"
+```
+
+
+### `io.buildpacks` (optional)
+
+This is the Cloud Native Buildpacks' section of the project descriptor. The TOML schema is the following:
+
+```
+[io.buildpacks]
+builder = "<string>"
+include = ["<string>"]
+exclude = ["<string>"]
+
+[[io.buildpacks.group]]
+id = "<string>"
+version = "<string>"
+uri = "<string>"
+
+  [io.buildpacks.group.script]
+  api = "<buildpack api>"
+  shell = "<string (optional default=/bin/sh)>"
+  inline = "<script contents>"
+
+[[io.buildpacks.build.env]]
```

Comment:
This has `io.buildpacks.build.env` vs the definition below has `io.buildpacks.env.build`. Which one should it be? I believe `io.buildpacks.build.env` makes more sense.

Reply:
Good catch, I'll update the spec.